pwiz_tools/Skyline/Model/Lib/ViewLibraryPepInfo.cs

```diff
@@ -1,221 +1,373 @@
+                                                      IsotopeLabelType.light, true, null), null);
+                if (Key.IsSmallMoleculeKey)
+                {
+                    mods = null;
+                    return;
+                }
+
+                // Because the document modifications do not explain this peptide, a set of
+                // explicit modifications must be constructed, even if they are empty.
+                IList<ExplicitMod> staticModList = new List<ExplicitMod>();
+                IEnumerable<ModificationInfo> modList = GetModifications();
+                foreach (var modInfo in modList)
+                {
+                    var aa = modInfo.ModifiedAminoAcid;
+                    var smod = new StaticMod(@"temp",
+                                             aa != 'X' ? aa.ToString(CultureInfo.InvariantCulture) : string.Join(@",", AminoAcid.All),
+                                             null,
+                                             null,
+                                             LabelAtoms.None,
+                                             modInfo.ModifiedMass,
+                                             modInfo.ModifiedMass);
+                    var exmod = new ExplicitMod(modInfo.IndexMod, smod);
+                    staticModList.Add(exmod);
+                }
+
+                mods = new ExplicitMods(peptide, staticModList, new TypedExplicitModifications[0]);
+            }
+            else
+            {
+                // Create custom ion node for midas library
+                var precursor = Key.PrecursorMz.GetValueOrDefault();
+                var precursorMono = new TypedMass(precursor, MassType.Monoisotopic);
+                var precursorAverage = new TypedMass(precursor, MassType.Average);
+                var peptide = new Peptide(new CustomMolecule(precursorMono, precursorAverage, precursor.ToString(CultureInfo.CurrentCulture)));
+                transitionGroup = new TransitionGroupDocNode(new TransitionGroup(peptide, Adduct.EMPTY, IsotopeLabelType.light, true, null), null);
+                mods = new ExplicitMods(peptide, new ExplicitMod[0], new TypedExplicitModifications[0]);
+            }
+        }
+
+        // Computes each ModificationInfo for the given peptide and returns a 
+        // list of all modifications.
+        private IEnumerable<ModificationInfo> GetModifications()
+        {
+            IList<ModificationInfo> modList = new List<ModificationInfo>();
+            string sequence;
+            if (Key.LibraryKey is PeptideLibraryKey peptideLibraryKey)
+            {
+                sequence = peptideLibraryKey.ModifiedSequence;
+            }
+            else
+            {
+                return modList;
+            }
+            int iMod = -1;
+            for (int i = 0; i < sequence.Length; i++)
+            {
+                if (sequence[i] != '[')
```

Comment:
Seems hard to believe there is no other code in the product that does this already.

Reply:
There probably wasn't at the time this code has been written 12 years ago.